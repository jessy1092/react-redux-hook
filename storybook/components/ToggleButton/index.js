import React from 'react';
import { createStore } from 'redux';

import { Provider, useRedux } from '../../../src';

const store = createStore((state = { toggle: false }, action) => {
	if (action.type === 'TOGGLE') {
		return { toggle: !state.toggle };
	}

	return state;
});

const ToggleButton = () => {
	const [state, dispatch] = useRedux();

	return (
		<button onClick={() => dispatch({ type: 'TOGGLE' })}>
			{state.toggle ? 'Click to close' : 'Click to open'}
		</button>
	);
};

const RootComponent = () => (
	<Provider store={store}>
		<ToggleButton />
	</Provider>
);

export default RootComponent;
