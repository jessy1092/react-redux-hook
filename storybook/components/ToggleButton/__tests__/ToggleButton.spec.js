import React from 'react';
import { mount } from 'enzyme';

import ToggleButton from '../index.js';

test('should have default value: Click to open', () => {
	const component = mount(<ToggleButton />);
	expect(component.text()).toContain('Click to open');
});

// Create custom snapshot testing
test('should become "Click to close" after click', () => {
	const component = mount(<ToggleButton />);

	expect(component).toMatchSnapshot();

	// manually trigger the callback
	component.find('button').simulate('click');

	// enzyme not support sideEffect yet
	// expect(component.text()).toContain('Click to close');
	// // re-rendering would become Open
	// expect(component).toMatchSnapshot();

	// // manually trigger the callback
	// component.find('button').simulate('click');

	// expect(component.text()).toContain('Click to open');
	// // re-rendering would become Close
	// expect(component).toMatchSnapshot();
});
