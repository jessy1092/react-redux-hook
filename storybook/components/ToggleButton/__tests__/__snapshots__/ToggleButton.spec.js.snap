// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should become "Click to close" after click 1`] = `
<RootComponent>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ToggleButton>
      <button
        onClick={[Function]}
      >
        Click to open
      </button>
    </ToggleButton>
  </Provider>
</RootComponent>
`;
